### PR TITLE
LPS-166596 Fix service builder backward compatibility in older 7.4 portal version

### PR DESCRIPTION
**Pull request by julschong**
resubmitted from #2372

models return null instead of StringPool.BLANK
this prevents many petra-string inclusion to build.gradle

### CODE REVIEW

**Comment by catarineaguiar** [comment]
@julschong semver errors:

```
 [exec] com.liferay.semantic.versioning.SemanticVersioningTest > testSemanticVersioning[/apps/segments/segments-api] FAILED
     [exec]     java.lang.AssertionError:   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec]     = ================================================== ========== ========== ========== ========== ==========
     [exec]     * com.liferay.segments.model                         MINOR      2.0.0      2.0.0      2.1.0      VERSION INCREASE REQUIRED
```
```
 [exec] com.liferay.semantic.versioning.SemanticVersioningTest > testSemanticVersioning[/apps/subscription/subscription-api] FAILED
     [exec]     java.lang.AssertionError:   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec]     = ================================================== ========== ========== ========== ========== ==========
     [exec]     * com.liferay.subscription.model                     MINOR      3.0.0      3.0.0      3.1.0      VERSION INCREASE REQUIRED
```
full log [here](https://testray.liferay.com/reports/production/logs/2022-11/test-1-28/test-portal-acceptance-pullrequest(master)/5097/modules-semantic-versioning-jdk8/0/2/jenkins-console.txt.gz?authuser=0)

```
[exec] com.liferay.semantic.versioning.SemanticVersioningTest > testSemanticVersioning[/apps/commerce/commerce-api] FAILED
     [exec]     java.lang.AssertionError:   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec]     = ================================================== ========== ========== ========== ========== ==========
     [exec]     * com.liferay.commerce.model                         MINOR      12.2.0     12.2.0     12.3.0     VERSION INCREASE REQUIRED
```
```
 [exec] com.liferay.semantic.versioning.SemanticVersioningTest > testSemanticVersioning[/apps/commerce/commerce-price-list-api] FAILED
     [exec]     java.lang.AssertionError:   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec]     = ================================================== ========== ========== ========== ========== ==========
     [exec]     * com.liferay.commerce.price.list.model              MINOR      6.1.0      6.1.0      6.2.0      VERSION INCREASE REQUIRED
```
```
[exec] com.liferay.semantic.versioning.SemanticVersioningTest > testSemanticVersioning[/apps/commerce/commerce-tax-api] FAILED
     [exec]     java.lang.AssertionError:   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec]     = ================================================== ========== ========== ========== ========== ==========
     [exec]     * com.liferay.commerce.tax.model                     MINOR      3.0.0      3.0.0      3.1.0      VERSION INCREASE REQUIRED
```
```
 [exec] com.liferay.semantic.versioning.SemanticVersioningTest > testSemanticVersioning[/apps/commerce/commerce-inventory-api] FAILED
     [exec]     java.lang.AssertionError:   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec]     = ================================================== ========== ========== ========== ========== ==========
     [exec]     * com.liferay.commerce.inventory.model               MINOR      6.1.0      6.1.0      6.2.0      VERSION INCREASE REQUIRED
```
```
 [exec] com.liferay.semantic.versioning.SemanticVersioningTest > testSemanticVersioning[/apps/commerce/commerce-product-api] FAILED
     [exec]     java.lang.AssertionError:   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec]     = ================================================== ========== ========== ========== ========== ==========
     [exec]     * com.liferay.commerce.product.model                 MINOR      19.0.0     19.0.0     19.1.0     VERSION INCREASE REQUIRED
```
```
     [exec] com.liferay.semantic.versioning.SemanticVersioningTest > testSemanticVersioning[/apps/document-library/document-library-sync-api] FAILED
     [exec]     java.lang.AssertionError:   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec]     = ================================================== ========== ========== ========== ========== ==========
     [exec]     * com.liferay.document.library.sync.model            MINOR      3.0.0      3.0.0      3.1.0      VERSION INCREASE REQUIRED
```
full log [here](https://testray.liferay.com/reports/production/logs/2022-11/test-1-28/test-portal-acceptance-pullrequest(master)/5097/modules-semantic-versioning-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

```
     [echo] ## /opt/dev/projects/github/liferay-portal/baseline-reports/portal-kernel.log
     [echo] ##
     [echo] 
[beanshell]   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
[beanshell] = ================================================== ========== ========== ========== ========== ==========
[beanshell] * com.liferay.announcements.kernel.model             MINOR      4.0.0      4.0.0      4.1.0      VERSION INCREASE REQUIRED
```
full log [here](https://testray.liferay.com/reports/production/logs/2022-11/test-1-28/test-portal-acceptance-pullrequest(master)/5097/semantic-versioning-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
test after rerunning servicebuilder

**Comment by tinatian** [comment]
@julschong Please make sure you remove QA related to flag to trigger a new test result scanning.

**Comment by zhangyan0701** [comment]
Semantic versioning errors in several modules.
Examples:
```
[beanshell] * com.liferay.announcements.kernel.model             MINOR      4.0.0      4.0.0      4.1.0      VERSION INCREASE REQUIRED
[beanshell] 	<   class      com.liferay.announcements.kernel.model.AnnouncementsDeliveryWrapper
[beanshell] 		+   method     toXmlString()
```
```
[beanshell] * com.liferay.asset.kernel.model                     MINOR      9.0.0      9.0.0      9.1.0      VERSION INCREASE REQUIRED
[beanshell] 	<   class      com.liferay.asset.kernel.model.AssetCategoryWrapper
[beanshell] 		+   method     toXmlString()
```
```
[beanshell] * com.liferay.document.library.kernel.model          MINOR      7.0.0      7.0.0      7.1.0      VERSION INCREASE REQUIRED
[beanshell] 	<   class      com.liferay.document.library.kernel.model.DLFileEntryMetadataWrapper
[beanshell] 		+   method     toXmlString()
```
Full console log:
https://testray.liferay.com/reports/production/logs/2022-11/test-1-31/test-portal-acceptance-pullrequest(master)/5856/semantic-versioning-jdk8/0/0/jenkins-console.txt.gz?authuser=0

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
rebased and reran ant build-services for semver issue

**Comment by julschong** [comment]
rebased and reran ant build-services for semantic issue

**Comment by zhangyan0701** [comment]
No regressions found.

**Comment by tinatian** [comment]
@julschong This pull had code conflicts with BChan's master, need to wait for his local changes merged in and resend.

**Comment by tinatian** [comment]
@julschong Please keep an eye on head of master, and rebase this pull

**Comment by julschong** [comment]
@tinatian I'd like to run ci test again to make sure there's no semver issue first, taking off QA pass tag now

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

File: modules/apps/object/object-api/bnd.bnd [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Bundle-Name: Liferay Object API
 Bundle-SymbolicName: com.liferay.object.api
-Bundle-Version: 46.0.1
+Bundle-Version: 47.0.0
```
Comment (julschong) [inline modules/apps/object/object-api/bnd.bnd:3]: major version bump caused by upstream

**Comment by julschong** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
ci:test:relevant

**Comment by zhangyan0701** [comment]
ci:test:relevant

**Comment by tinatian** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/126430

**Pull request closed without merging**